### ISSUE TITLE
Hangfire.Console with Redis does not seems to work

### ISSUE THREAD

**Issue by JustinAren** [pieceofsummer/Hangfire.Console#32]
Hi,

I want to use Hangfire.Console with Redis, but it does not seems to work.

I have the following configuration code:
```csharp
GlobalConfiguration.Configuration
    .UseRedisStorage(RedisMultiplexer)
    .UseConsole();
```
The JobMethod:
```csharp
public void AddLatestDocsToIndex(PerformContext context)
{
    var lastId = this.GetIdsByIndexerConfiguration(this.IndexerConfigurations["GetLastDocIdFromIndex"]).FirstOrDefault();
    var models = TransactionEntityModel.GetBiggerThanId(lastId);
    if (!models.Any())
    {
        context.WriteLine("No new documents found.");
	return;
    }
    var helper = new SearchHelper<TransactionEntityConfiguration, TransactionEntityModel>();
    helper.AddOrUpdateBulk(models);
    context.WriteLine($"Added {models.Count} document(s) to the index.");
}
```
And the Hangfire recurring job call:
```csharp
RecurringJob.AddOrUpdate<TransactionEntityConfiguration>(
    "TransactionLatestDocsIndexer",
    x => x.AddLatestDocsToIndex(null),				
    Cron.MinuteInterval(5));
```

In the dashboard I don't see any Logging message. Also, when I look in Redis, the log messages don't show up anywhere.

I am using the following packages and versions:
```csharp
<package id="Hangfire" version="1.6.12" targetFramework="net451" />
<package id="Hangfire.Console" version="1.3.1" targetFramework="net451" />
<package id="Hangfire.Core" version="1.6.12" targetFramework="net451" />
<package id="Hangfire.Redis.StackExchange" version="1.6.7.24" targetFramework="net451" />
```
Am I doing something wrong?

**Comment by pieceofsummer** [comment]
Unfortunately, I don't have a Redis instance to check it. Neither I have any experience with it.

I'd suggest you to check your configuration first:
1. Do you have Server and Dashboard running in a single process? If they're running separately, you need to call `UseConsole()` on both.
2. Check if `PerformContext` passed to your job method is not `null`. Console will silently ignore `WriteLine`s called on a null context.

**Comment by JustinAren** [comment]
1. `UseConsole()` is called on both server and dashboard.
2. How do I check if `PerformContext` is not `null`? As I understand you have to call the enqueue function with `null` as `PerformContext`, so when I'm debugging the function, the `PerformContext` is `null`, but I don't know how to debug in Hangfire and to check how the `PerformContext` looks like.

Do you know anyone with a Redis instance where you can test the Hangfire.Console?

**Comment by alecgorge** [comment]
I just wanted to jump in and say that it wasn't not working for me either after I updated all my packages to the latest versions.

Switching to the Postgres store fixed everything. I don't intend on switching back but I wanted to chime in and say I was experiencing this as well. I was only using Redis while the Postgres store didn't function correctly.

**Comment by unicrus** [comment]
@pieceofsummer, Redis can be installed locally even on Windows: https://github.com/MSOpenTech/redis/releases.

I just tried Hangfire.Console on it and the console is displayed when task is running. But it disappears on finish.

Here is the used packages:
```
<package id="Hangfire.Console" version="1.3.1" targetFramework="net461" />
<package id="Hangfire.Core" version="1.6.5" targetFramework="net461" />
<package id="Hangfire.Redis.StackExchange" version="1.6.7.24" targetFramework="net461" />
```

**Comment by pieceofsummer** [comment]
Fixed in v1.3.2.

Redis storage implementation doesn't follow other providers' convention to return negative TimeSpan for non-expiring records, as it returns zero TimeSpan instead. I've added a workaround for such cases.

**P.S.** Is it just me, or Hangfire.Redis.StackExchange returns job states in reverse order?
